### PR TITLE
letsencrypt: Added support for Simply.com DNS challenge.

### PR DESCRIPTION
**Pull request by JohNan**
Added support for Simply.com DNS challenge.

## Summary by CodeRabbit

- **New Features**
	- Added support for Simply DNS, enhancing compatibility with DNS providers for certificate management.
	- Introduced configuration parameters for Simply API, allowing users to manage certificates using Simply DNS.

- **Updates**
	- Version incremented to 5.1.1, reflecting the new release.
	- Updated DNS provider options to include `dns-simply`.
	- Added detailed documentation for configuring Simply DNS challenges, including examples and credential clarifications.

### CODE REVIEW

**Comment by JohNan** [comment]
@coderabbitai review

**Comment by agners** [comment]
Thanks for your contribution! Generally LGTM.

Since the earlier PR #3716 just got merged, this needs a rebase and another version bump.

**Comment by JohNan** [comment]
@coderabbitai review

File: letsencrypt/DOCS.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Home Assistant Add-on: Letsencrypt
+O# Home Assistant Add-on: Letsencrypt
```
Comment (agners) [inline letsencrypt/DOCS.md:1]: Typo?

```suggestion
# Home Assistant Add-on: Letsencrypt
```

**Review by agners — APPROVED** [review]

File: letsencrypt/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## 5.1.1
+
+- Add Simply DNS support
```
[the remark this answers is not in the document]
Comment (agners) [reply letsencrypt/CHANGELOG.md:5]: Already got merged :cry: . But we can change the changelog still, simply create a PR :sweat_smile:

**Pull request merged**